core/src/main/java/com/datastax/oss/driver/internal/core/pool/ChannelSet.java

```diff
@@ -83,18 +83,25 @@ DriverChannel next() {
       case 0:
         return null;
       case 1:
-        return snapshot[0];
+        DriverChannel onlyChannel = snapshot[0];
+        return onlyChannel.preAcquireId() ? onlyChannel : null;
       default:
-        DriverChannel best = null;
-        int bestScore = 0;
-        for (DriverChannel channel : snapshot) {
-          int score = channel.getAvailableIds();
-          if (score > bestScore) {
-            bestScore = score;
-            best = channel;
+        while (true) {
```

Comment:
Maybe we should add a limit to the number of iterations, to avoid a starvation condition if a thread consistently sees channels with available ids, but the best one always fills up by the time it tries to acquire.

Reply:
After testing locally, I think we are pretty safe. Even with a high level of contention (1000 threads competing for 5 connections with 20 requests/connection), the highest number of iterations I've observed is 3, and this happens 5 times over a 2-minute test period. One iteration in the loop usually takes about 200ns.